coprocessor/fhevm-engine/host-listener/src/database/tfhe_event_propagate.rs

```diff
@@ -663,19 +690,67 @@ impl Database {
     async fn record_transaction_begin(
         &self,
         transaction_hash: &Option<Vec<u8>>,
-        block_number: &Option<u64>,
+        block_number: u64,
     ) {
         if let Some(txn_id) = transaction_hash {
             let pool = self.pool.read().await.clone();
             let _ = telemetry::try_begin_transaction(
                 &pool,
                 self.chain_id as i64,
                 txn_id.as_ref(),
-                block_number.unwrap_or_default(),
+                block_number,
             )
             .await;
         }
     }
+
+    #[allow(clippy::too_many_arguments)]
+    async fn insert_delegation(
+        tx: &mut Transaction<'_>,
+        delegator: Address,
+        delegate: Address,
+        contract_address: Address,
+        delegation_counter: u64,
+        old_expiration_date: u64,
+        new_expiration_date: u64,
+        chain_id: u64,
+        block_hash: &[u8],
+        block_number: u64,
+        transaction_id: Option<Vec<u8>>,
+    ) -> Result<(), SqlxError> {
+        // ON CONFLIT is done on Unique constraint
+        let query = sqlx::query!(
+            "INSERT INTO delegate_user_decrypt(
+                delegator, delegate, contract_address, delegation_counter, old_expiration_date, new_expiration_date, host_chain_id, block_number, block_hash, transaction_id, on_gateway, reorg_out)
+            VALUES($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, false, false)
+            ON CONFLICT DO NOTHING",
+            &delegator.into_array(),
+            &delegate.into_array(),
+            &contract_address.into_array(),
+            delegation_counter as i64,
+            old_expiration_date as i64,
+            new_expiration_date as i64,
+            chain_id as i64,
+            block_number as i64,
```

Comment:
Not sure this is safe, casting u64 to i64?
Not a problem now but might become one later.

Reply:
Depend on the use case. (e.g. cast and cast back is safe).
For which value do you think it's risky ? timestamp, chain id block, number ?